src/Mod/TechDraw/Gui/ViewMakers.cpp

```diff
@@ -0,0 +1,491 @@
+{
+    std::string FeatName = page->getDocument()->getUniqueObjectName("View");
+    std::string PageName = page->getNameInDocument();
+
+    //NOLINTBEGIN
+    Gui::Command::doCommand(Gui::Command::Doc, "App.activeDocument().addObject('TechDraw::DrawProjGroupItem', '%s')",
+                   FeatName.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "App.activeDocument().%s.translateLabel('DrawProjGroupItem', 'View', '%s')",
+                   FeatName.c_str(), FeatName.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "App.activeDocument().%s.addView(App.activeDocument().%s)", PageName.c_str(),
+                   FeatName.c_str());
+    //NOLINTEND
+
+    App::DocumentObject* docObj = page->getDocument()->getObject(FeatName.c_str());
+    auto* dvp = dynamic_cast<TechDraw::DrawViewPart*>(docObj);
+    if (!dvp) {
+        throw Base::TypeError("CmdTechDrawView DVP not found\n");
+    }
+    dvp->Source.setValues(shapeObjs);
+    dvp->XSource.setValues(xShapeObjs);
+
+//    cmd->getDocument()->setStatus(App::Document::Status::SkipRecompute, true);
+
+    //NOLINTBEGIN
+    auto dirs = CommandHelpers::viewDirection();
+    Gui::Command::doCommand(Gui::Command::Doc,
+                   "App.activeDocument().%s.Direction = FreeCAD.Vector(%.12f, %.12f, %.12f)",
+                   FeatName.c_str(), dirs.first.x, dirs.first.y, dirs.first.z);
+    Gui::Command::doCommand(Gui::Command::Doc,
+                   "App.activeDocument().%s.RotationVector = FreeCAD.Vector(%.12f, %.12f, %.12f)",
+                   FeatName.c_str(), dirs.second.x, dirs.second.y, dirs.second.z);
+    Gui::Command::doCommand(Gui::Command::Doc,
+                   "App.activeDocument().%s.XDirection = FreeCAD.Vector(%.12f, %.12f, %.12f)",
+                   FeatName.c_str(), dirs.second.x, dirs.second.y, dirs.second.z);
+    //NOLINTEND
+
+//    cmd->getDocument()->setStatus(App::Document::Status::SkipRecompute, false);
+
+    return dvp;
+}
+
+//! returns a DrawProjGroup based on the current selection.
+DrawProjGroup* ViewMakers::makeProjectionGroupSelection(DrawPage* page)
+{
+    std::vector<App::DocumentObject*> shapeObjs;
+    std::vector<App::DocumentObject*> xShapeObjs;
+    CommandHelpers::getShapeObjectsFromSelection(shapeObjs, xShapeObjs);
+    auto directions = CommandHelpers::viewDirection();
+    return makeProjectionGroup(page, shapeObjs, xShapeObjs, directions);
+}
+
+//! returns a DrawProjGroup based on parameters.
+DrawProjGroup* ViewMakers::makeProjectionGroup(DrawPage* page,
+                                          const std::vector<App::DocumentObject*>& shapes,
+                                          const std::vector<App::DocumentObject*>& xShapes,
+                                          const std::pair<Base::Vector3d, Base::Vector3d>& directions)
+{
+    std::string PageName = page->getNameInDocument();
+    std::string multiViewName = page->getDocument()->getUniqueObjectName("ProjGroup");
```

Comment:
Creation of views are GUI independent, so shouldn't it be in App?

Reply:
There used to be a TODO comment that said the makexxxxxx methods should go to App and the makexxxxxxSelections methods should stay in Gui.  Don't know where that comment went.